catalog/core/catalog-core-api/src/main/java/ddf/catalog/content/encryption/EncryptorDecryptor.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.content.encryption;
+
+import java.io.InputStream;
+
+/** Interface for encrypting and decrypting data */
+public interface EncryptorDecryptor {
```

Comment:
I'm going to leave it as is for now. My idea of the `EncryptorDecryptor` interface is that it can encrypt data and be able to decrypt the same data it has encrypted. This fits the use case of the `FileSystemStorageProvider`. 

And as you mentioned, we _could_ separate those out into two separate interfaces, but I would argue that is more complicated than what we need at the moment. And I am a sucker for over-engineering, so I'm going to show some self control for once 😆.